test/metabase/db/metadata_queries_test.clj

```diff
@@ -74,6 +74,21 @@
             (let [query (#'metadata-queries/table-rows-sample-query table fields {:truncation-size 4})]
               (is (empty? (get-in query [:query :expressions]))))))))))
 
+(deftest mbql-on-table-requires-filter-will-include-the-filter-test
+  (mt/with-temp
+    [:model/Database db     {}
+     :model/Table    table  {:database_require_filter true :db_id (:id db)}
+     :model/Field    field1 {:name "name" :table_id (:id table)}
```

Comment:
I think you should add the type here explicitly since you rely on it on line 85.
```suggestion
     :model/Field    field1 {:name "name" :table_id (:id table) :base_type :type/Integer}
```